### ISSUE TITLE
Analyzer error around incorrect types is wrong

### ISSUE THREAD

**Issue by matanlurey** [dart-lang/sdk#33946]
Internal issue: [b/111717399](b/111717399).

Example error:

```
The argument type 'List<Queue> (/tmp/.../dart_lang/v2_0_0_dev_64_2/sdk/lib/core/list.dart)' can't be assigned to the parameter type 'List<Queue> (/tmp/.../dart_lang/v2_0_0_dev_64_2/sdk/lib/core/list.dart)'. #argument_type_not_assignable
```

As noted by @davidmorgan, this is caused if you do something like:

```dart
// my_queue.dart
class Queue {}
```

```dart
// main.dart
import 'dart:collection' as lib1 show Queue;
import 'my_queue.dart' as lib2 show Queue;

void main() {
  List<lib1.Queue> list = new List<lib2.Queue>();
}
```

The error message should have highlighted that one of the `Queue` implementations came from `my_queue.dart`, but does not.

**Comment by matanlurey** [comment]
Ping as I think this fell off / was not triaged.

**Comment by matanlurey** [comment]
Ping. How can we get this prioritized and fixed?

**Comment by davidmorgan** [comment]
Maybe @bwilkerson ?

**Comment by davidmorgan** [comment]
Not sure about priority labels, but the problem is that the error message in these cases in misleading and has led to lost time.

**Comment by matanlurey** [comment]
Internally the bug that was filed was P1 as it took multiple Dart engineers to figure out what was going on and wasted a lot of time.

**Comment by bwilkerson** [comment]
Thanks. I agree that this is a serious UX problem that needs to be fixed. Unfortunately, as frustrating as it is for me personally, I don't know when we will have the resources to address this issue.

**Comment by matanlurey** [comment]
@devoncarew: How can we get resources to address this issue? I imagine its relatively simple (~minutes for someone who understands the code base), and would *greatly* help usability. 

Even just pointing out _where_ the code that produces this error message is would be very helpful, we could solicit other folks to try and land the actual fix.

**Comment by devoncarew** [comment]
@matanlurey, we have a few more cycles available for 2.1 work now and are triaging what we can do in that timeframe; I'll keep this issue in the Dart2.1 milestone.

**Comment by bwilkerson** [comment]
The message formatting isn't ideal, but we at least point users to the correct types as of https://dart-review.googlesource.com/c/sdk/+/87425.
